main.cpp

```diff
@@ -25,69 +25,58 @@ Create a branch named Part2
 #include <string>
 struct T
 {
-    T(int v, const char* l) : value(v), name(l)  {}  //1
+    T(int v, const std::string& l) : value(v), name(l)  {}  //1
     int value; //2
     std::string name; //3
 };
 struct Compare                                //4
 {
-    T* compare(T* a, T* b) //5
+    T* compare(T& a, T& b) //5
     {
-        if (a != nullptr && b != nullptr)
-        {
-            if( a->value < b->value ) return a;
-            if( a->value > b->value ) return b;
-        }
-        return nullptr;
+        if( a.value < b.value ) return &a;
+        if( a.value > b.value ) return &b;
+        return nullptr; 
     }
 };
 
 struct U
 {
     float number1 { 0 }, number2 { 0 };
-    float distanceShrinker(float* valueUpdated)      //12
+    float distanceShrinker(float& valueUpdated)      //12
     {
-        if (valueUpdated != nullptr)
+        
+        
+        std::cout << "U's number1 value: " << number1 << std::endl;
```

Comment:
When you delete code, you need to clean up the blank lines that result from that deletion
- [ ] revise